recipes/tls-create-secure-pair-to-tls-socket/src/workflow.ts

```diff
@@ -0,0 +1,236 @@
+import type JS from '@codemod.com/jssg-types/langs/javascript';
+import { getNodeImportStatements } from '@nodejs/codemod-utils/ast-grep/import-statement';
+import { getNodeRequireCalls } from '@nodejs/codemod-utils/ast-grep/require-call';
+import { resolveBindingPath } from '@nodejs/codemod-utils/ast-grep/resolve-binding-path';
+
+type CallSite = { call: SgNode<JS>; binding: string };
+
+export default function transform(root: SgRoot<JS>): string | null {
+	const rootNode = root.root();
+	const tlsStmts = [
+		...getNodeImportStatements(root, 'tls'),
+		...getNodeRequireCalls(root, 'tls'),
+	];
+	const cspBindings = unique([
+		...tlsStmts.map(s => resolveBindingPath(s as unknown as SgNode<JS>, '$.createSecurePair')).filter(Boolean) as string[],
+		...collectDefaultImportBindings(tlsStmts),
+		...collectDynamicImportIdentifiers(rootNode),
+	]);
+	const callSites: CallSite[] = [...findCreateSecurePairCalls(rootNode)];
+	const edits: Edit[] = [];
+	for (const { call, binding } of callSites) {
+		const a = getText(call.getMatch('A'));
+		const b = getText(call.getMatch('B'));
+		const c = getText(call.getMatch('C'));
+		const d = getText(call.getMatch('D'));
+		const options = buildOptions(a, b, c, d);
+		const isNamespace = binding.includes('.');
+		const replacement = isNamespace
+			? `new ${binding.replace(/\.createSecurePair$/, '.TLSSocket')}(underlyingSocket, ${options})`
+			: `new TLSSocket(underlyingSocket, ${options})`;
+		edits.push(call.replace(replacement));
+	}
+	edits.push(...renamePairAssignedVariables(rootNode, cspBindings));
+	edits.push(...rewriteTlsImports(rootNode));
+	if (edits.length === 0) return null;
+	return rootNode.commitEdits(edits);
+}
+
+function findCreateSecurePairCalls(rootNode: SgNode<JS>): CallSite[] {
+	const pats = [
+		'$X.createSecurePair($A, $B, $C, $D)',
+		'$X.createSecurePair($A, $B, $C)',
+		'$X.createSecurePair($A, $B)',
+		'$X.createSecurePair($A)',
+		'$X.createSecurePair()',
+		'createSecurePair($A, $B, $C, $D)',
+		'createSecurePair($A, $B, $C)',
+		'createSecurePair($A, $B)',
+		'createSecurePair($A)',
+		'createSecurePair()',
+	];
+	const out: CallSite[] = [];
+	for (const p of pats) {
+		const nodes = rootNode.findAll({ rule: { kind: 'call_expression', pattern: p } });
+		for (const n of nodes) {
+			const x = (n as SgNode<JS>).getMatch('X')?.text();
+			const binding = x ? `${x}.createSecurePair` : 'createSecurePair';
+			out.push({ call: n as SgNode<JS>, binding });
+		}
+	}
+	return out;
+}
+
+function buildOptions(a?: string | null, b?: string | null, c?: string | null, d?: string | null): string {
+	const kv: string[] = [];
+	if (a) kv.push(`secureContext: ${a}`);
+	if (b) kv.push(`isServer: ${b}`);
+	if (c) kv.push(`requestCert: ${c}`);
+	if (d) kv.push(`rejectUnauthorized: ${d}`);
+	return `{ ${kv.join(', ')} }`;
+}
+
+function getText(node: SgNode<JS> | undefined): string | null {
+	const t = node?.text()?.trim();
+	return t || null;
+}
```

Comment:
I think you don't really want to literally pass `undefined` (`getText(undefined)`, you just want `node` to be optional (`getText(maybeUndefined)`).

```suggestion
function getText(node?: SgNode<JS>) {
	return node?.text()?.trim() || null;
}
```

Why return `null` at all though instead of just an empty string or undefined?